crates/rust-project-goals-cli/src/cfp.rs

```diff
@@ -0,0 +1,533 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use anyhow::{Context, Result};
+use regex::Regex;
+
+/// Module containing pure functions for text processing
+pub mod text_processing {
+    use regex::Regex;
+    
+    /// Process template content by replacing placeholders and removing notes
+    pub fn process_template_content(content: &str, timeframe: &str, lowercase_timeframe: &str) -> String {
+        // Remove note sections (starting with '> **NOTE:**' and continuing until a non-'>' line)
+        // But preserve other content starting with '>'
+        let lines: Vec<&str> = content.lines().collect();
```

Comment:
An example of things we can fix later. The loop should use the iterator directly.